### PR TITLE
Use more java 8 features: Replace Collections.sort() with List.sort()

### PR DESCRIPTION
**Pull request by pavog**
Uses List.sort() instead of Collections.sort() to sort lists.
This function was introduced with Java 8 and might speed up sorting a bit.

See [javadocs](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#sort-java.util.Comparator-) for reference.

### CODE REVIEW

**Comment by pavog** [comment]
Oops, I got the function names mixed up in the commit message.

**Comment by kaikreuzer** [comment]
Please do not do any merge commits on a PR - rather rebase instead.

**Comment by pavog** [comment]
Oops, sorry, I didn't realize.
Can I undo this or should I create a new PR?

**Comment by cweitkamp** [comment]
@pavog You do not need to create a new PR. You can squash your commits and force push afterwards to your branch.

File: bundles/org.openhab.core.io.http/src/main/java/org/openhab/core/io/http/internal/SmartHomeHttpContext.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public HttpContext wrap(Bundle bundle) {
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     public void addHandler(Handler handler) {
         this.handlers.add(handler);
-        Collections.sort(handlers, Comparator.comparingInt(Handler::getPriority));
+        handlers.sort(Comparator.comparingInt(Handler::getPriority));
```
Comment (cweitkamp) [inline bundles/org.openhab.core.io.http/src/main/java/org/openhab/core/io/http/internal/SmartHomeHttpContext.java:85]: I guess you now can remove the import of `java.util.Collections` in this class too.

**Comment by pavog** [comment]
@cweitkamp Thank you very much for your help! I didn't want to mess up more things, so I created a new pull request at #1564.

**Pull request closed without merging**